spec/components/email_updater_spec.rb

```diff
@@ -17,6 +17,92 @@
     expect(updater.errors.messages[:base].first).to be I18n.t("change_email.error_staged")
   end
 
+  context "when an admin is changing the email of a user" do
+    let(:email_change_requesting_user) { Fabricate(:admin) }
+    let(:updater) { EmailUpdater.new(user.guardian, user, email_change_requesting_user) }
+
+    def expect_old_email_job
+      Jobs.expects(:enqueue).with(:critical_user_email, has_entries(to_address: "old.email@example.com", type: :notify_old_email, user_id: user.id))
+    end
+
+    def expect_forgot_password_job
+      Jobs.expects(:enqueue).with(:critical_user_email, has_entries(type: :forgot_password, user_id: user.id))
+    end
+
+    context "for a regular user" do
+      let(:user) { Fabricate(:user, email: old_email) }
+
+      it "does not send an email to the user for them to confirm their new email but still sends the notification to the old email" do
+        Jobs.expects(:enqueue).with(:critical_user_email, has_entries(type: :confirm_new_email, to_address: new_email)).never
+        expect_old_email_job
+        expect_forgot_password_job
+        updater.change_to(new_email)
+      end
+
+      it "creates a change request authorizing the new email and immediately confirms it " do
+        updater.change_to(new_email)
+        change_req = user.email_change_requests.first
+        expect(change_req.change_state).to eq(EmailChangeRequest.states[:complete])
+      end
+
+      it "sends a reset password email to the user so they can set a password for their new email" do
+        expect_old_email_job
+        expect_forgot_password_job
+        updater.change_to(new_email)
+      end
+    end
+
+    context "for a staff user" do
+      let(:user) { Fabricate(:moderator, email: old_email) }
```

Comment:
Calling this `moderator` makes it easier to read

```suggestion
      let(:moderator) { Fabricate(:moderator, email: old_email) }
```